app/services/transfers/service.go

```diff
@@ -329,6 +331,9 @@ func (ts *Service) TransferData(txId string) (service.TransferData, error) {
 		ts.logger.Errorf("[%s] - Failed to query Transfer with messages. Error: [%s].", txId, err)
 		return service.TransferData{}, err
 	}
+	if t == nil || t.Fee.Amount == "" {
+		return service.TransferData{}, nil
+	}
```

Comment:
I think that it would be better if we return `404` in the `transfer` router if the transfer is not found